crypto/s2n_openssl_x509.c

```diff
@@ -110,12 +110,22 @@ S2N_RESULT s2n_openssl_x509_get_cert_info(X509 *cert, struct s2n_cert_info *info
 #else
     info->signature_nid = X509_get_signature_nid(cert);
 #endif
-    /* These is no method to directly retrieve that signature digest from the X509*
+
+    /* There is no method to directly retrieve the signature digest from the X509*
      * that is available in all libcryptos, so instead we use find_sigid_algs. For
-     * a signature NID_ecdsa_with_SHA256 this will return NID_SHA256 
+     * a signature with NID_ecdsa_with_SHA256 this will return NID_SHA256
+     *
+     * signature_digest_nid may not always be set. ML-DSA does not have an associated digest.
+     *
+     * Note: this method may fail for ML-DSA-44 and ML-DSA-87, depending on the
+     * version of AWS-LC. See https://github.com/aws/aws-lc/issues/2347
+     * In order to handle this bug, we intentionally do not check the return
+     * value of OBJ_find_sigid_algs. Since OBJ_find_sigid_algs should only fail
+     * if signature_nid is unsupported by the libcrypto, and we retrieve signature_nid
+     * from the libcrypto, OBJ_find_sigid_algs should only fail for library bugs
+     * like the linked issue.
      */
-    RESULT_GUARD_OSSL(OBJ_find_sigid_algs(info->signature_nid, &info->signature_digest_nid, NULL),
-            S2N_ERR_CERT_TYPE_UNSUPPORTED);
+    OBJ_find_sigid_algs(info->signature_nid, &info->signature_digest_nid, NULL);
```

Comment:
I've got a PR out to fix the awslc bug: https://github.com/aws/aws-lc/pull/2348

But even if I fix it, we won't be able to assume that we're building against a fixed version of awslc :/ So I think we're stuck with handling this case. If just ignoring the error isn't acceptable, I could instead skip the method call for the MLDSA nids. That's a pretty ugly check though: I'll need to specifically list the MLDSA nids to skip. And it'll need an if/def since the nids don't exist in awslc before MLDSA support was added.

We could also ask awslc to bump their API version, and then gate MLDSA support on the newer API version.

Reply:
Hmmm so like check for failure, but if the failure occurs, just explicitly set signature_digest_nid to 0 in case OBJ_find_sigid_algs set it to something else? I don't think that's super useful, but I'm probably biased by having looked at the implementation of OBJ_find_sigid_algs :P I can do that.

Reply:
Okay so the current plan:
1. Initially, follow Sam's suggestion and reset signature_digest_nid on failure, but otherwise ignore failures.
2. Update AWSLC's API version: https://github.com/aws/aws-lc/pull/2349
3. Wait for the next AWSLC release (probably later this week)
4. Rebuild our codebuild images
5. Make ML-DSA support depend on the new API version and revert the changes to this logic.